Makefile

```diff
@@ -21,6 +21,16 @@ LEADER = controls-leader
 MOTORSIM = motor-sim
 CARSIM = car-sim
 
+# TODO: Remove BSP option once HAL is fully integrated
+# --------------------------------------------------------------
+BSP ?= StdPeriph
+BSP_PATH = BSP/STM32F413
+
+ifeq ($(BSP), HAL)
+	BSP_PATH := Embedded-Sharepoint/$(BSP_PATH)
+endif
```

Comment:
Maybe add a flag to pass to the make command to switch to the HAL path?

Reply:
The BSP variable is the flag, so if BSP=StdPeriph then it will choose the existing BSP, if BSP=HAL it will choose the Embedded-Sharepoint version of it.

Reply:
Yeah, but I mean like making a flag to pass in to the make command like 'make -HAL stm32 ...' so  can use the hal without having to edit the variable in the file

Reply:
You can always run make BSP=HAL to set the BSP variable to HAL